service/matching/matchingEngine.go

```diff
@@ -165,8 +167,17 @@ func (e *matchingEngineImpl) String() string {
 
 // Returns taskListManager for a task list. If not already cached gets new range from DB and
 // if successful creates one.
-func (e *matchingEngineImpl) getTaskListManager(taskList *taskListID,
-	taskListKind *types.TaskListKind) (taskListManager, error) {
+func (e *matchingEngineImpl) getTaskListManager(
+	taskList *taskListID,
+	taskListKind *types.TaskListKind,
+) (taskListManager, error) {
+	// Cache user defined task list in taskListCache
+	if e.config.EnableTaskListCache() {
+		if *taskListKind == types.TaskListKindNormal {
+			e.cacheTaskListByDomain(taskList)
```

Comment:
Is the taskListsLock held when calling this func? Otherwise two goroutines may operated on the same cached map I think?

Reply:
Yes. Updated.